lms/static/js/instructor_dashboard/data_download_2.js

```diff
@@ -0,0 +1,402 @@
+/* globals _, DataDownloadV2, PendingInstructorTasks, ReportDownloads */
+
+(function() {
+    'use strict';
+  // eslint-disable-next-line no-unused-vars
+    var DataDownloadV2, PendingInstructorTasks, ReportDownloads, statusAjaxError;
+
+    statusAjaxError = function() {
+        return window.InstructorDashboard.util.statusAjaxError.apply(this, arguments);
+    };
+
+    PendingInstructorTasks = function() {
+        return window.InstructorDashboard.util.PendingInstructorTasks;
+    };
+
+    ReportDownloads = function() {
+        return window.InstructorDashboard.util.ReportDownloads;
+    };
+
+    DataDownloadV2 = (function() {
+        function InstructorDashboardDataDownload($section) {
+            var dataDownloadObj = this;
+            this.$section = $section;
+            this.$section.data('wrapper', this);
+            this.$list_problem_responses_csv_input = this.$section.find("input[name='problem-location']");
+            this.$download_display_text = $('.data-display-text');
+            this.$download_request_response_error = $('.request-response-error');
+            this.$download_display_table = $('.profile-data-display-table');
+            this.$reports_request_response = $('.request-response');
+            this.$reports_request_response_error = $('.request-response-error');
+            this.report_downloads = new (ReportDownloads())(this.$section);
+            this.instructor_tasks = new (PendingInstructorTasks())(this.$section);
+            this.$download_report = $('.download-report');
+            this.$gradeReportDownload = $('.grade-report-download');
+            this.$report_type_selector = $('.report-type');
+            this.$selection_informations = $('.selectionInfo');
+            this.$certificate_display_table = $('.certificate-data-display-table');
+            this.$downloadProblemReport = $('#download-problem-report');
+            this.$tabSwitch = $('.data-download-nav .btn-link');
+            this.$selectedSection = $('#' + this.$tabSwitch.first().attr('data-section'));
+            this.$learnerStatus = $('.learner-status');
+
+            this.clear_display = function() {
+                this.$download_display_text.empty();
+                this.$download_display_table.empty();
+                this.$download_request_response_error.empty();
+                this.$reports_request_response.empty();
+                this.$reports_request_response_error.empty();
+                this.$certificate_display_table.empty();
+                $('.msg-confirm').css({
+                    display: 'none'
+                });
+                return $('.msg-error').css({
+                    display: 'none'
+                });
+            };
+
```

Comment:
Why are the functions added directly? Why not via Prototype?

Reply:
`clear_display()` is called on initialization of this object and if it is added by Prototype it will be undefined at initialization of the object which gives an error.